lib/api/include/irods/check_auth_credentials.h

```diff
@@ -0,0 +1,134 @@
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed sizeof(UserInfo::rodsZone) - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    /// - It must not include the pound sign
+    /// - It must not include the username
+    ///
+    /// \since 4.3.1
+    char zone[64];
+
+    /// The expected obfuscated password for the user of interest.
+    ///
+    /// The following requirements must be satisfied:
+    /// - The length must not exceed 250 - 1 bytes
+    /// - It must be null-terminated
+    /// - It must be non-empty
+    ///
+    /// \since 4.3.1
+    char password[250];
+
+    // NOLINTEND(cppcoreguidelines-avoid-magic-numbers, readability-magic-numbers)
+    // NOLINTNEND(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+} checkAuthCredsInp_t;
+
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+#define CheckAuthCredentialsInput_PI "str username[64]; str zone[64]; str password[250];"
+
+/// The output data type used to invoke #rc_check_auth_credentials.
+///
+/// \since 4.3.1
+typedef struct CheckAuthCredentialsOutput // NOLINT(modernize-use-using)
+{
+    /// An integer indicating whether the credentials for a particular user are correct
+    /// or not.
+    ///
+    /// If the credentials are correct, this member will be set to 1. Otherwise, it is
+    /// set to 0.
+    int would_succeed;
+} checkAuthCredsOut_t;
+
+// NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
+#define CheckAuthCredentialsOutput_PI "int would_succeed;"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// Verifies the credentials passed match the credentials stored in the catalog.
+///
+/// This API endpoint only supports Native authentication and requires rodsadmin
+/// privileges.
+///
+/// It does NOT modify anything state on either side of the connection.
+///
+/// \param[in] _comm   A pointer to a RcComm.
+/// \param[in] _input  A pointer to a CheckAuthCredentialsInput.
+/// \param[in] _output A pointer to a CheckAuthCredentialsOutput.
+///
+/// \return An integer.
+/// \retval 0  On success.
+/// \retval <0 On failure.
+///
+/// \b Example
+/// \code{.cpp}
+/// RcComm* comm = // Our rodsadmin connection object.
+///
+/// // Configure the input object for the API call.
+/// struct CheckAuthCredentialsInput input;
+/// memset(&input, 0, sizeof(struct CheckAuthCredentialsInput));
+///
+/// strncpy(input.username, "bob", sizeof(input.username) - 1);
+/// strncpy(input.zone, "tempZone", sizeof(input.zone) - 1);
+///
+/// char* expected_password = // What we expect bob's password to be.
+/// char* scrambled_password = // expected_password, but scrambled. See #obfEncodeByKey.
+/// strncpy(input.password, scrambled_password, sizeof(input.password) - 1);
+///
+/// struct CheckAuthCredentialsOutput* output = NULL;
+///
+/// // Verify if the given password for bob is correct.
+/// const int ec = rc_check_auth_credentials(comm, &input, &output);
+///
+/// if (ec < 0) {
+///     // Handle error.
```

Comment:
Is it possible for `output` to be allocated but an error has occurred? It looks like it is not, but maybe worth mentioning somewhere?

Reply:
That is the intent. If an error occurs, `output` should be null.

Certainly worth mentioning in the documentation for the API.

Reply:
Done.